A6.md

```diff
@@ -0,0 +1,398 @@
+gRPC Retry Design
+----
+* Author(s): [Noah Eisen](https://github.com/ncteisen) and [Eric Gribkoff](https://github.com/ericgribkoff)
+* Approver: a11r
+* Status: Draft
+* Implemented in: All Languages
+* Last updated: 02/10/2017
+* Discussion at: https://groups.google.com/forum/#!topic/grpc-io/zzHIICbwTZE
+
+Table of Contents
+----
+  * [Abstract](#abstract)
+  * [Background](#background)
+  * [Proposal](#proposal)
+     * [Overview](#overview)
+     * [Detailed Design](#detailed-design)
+        * [Retry Policy Capabilities](#retry-policy-capabilities)
+           * [Maximum Number of Retries](#maximum-number-of-retries)
+           * [Exponential Backoff](#exponential-backoff)
+           * [Retryable Status Codes](#retryable-status-codes)
+        * [Hedging Policy](#hedging-policy)
+        * [Throttling Retry Attempts and Hedged RPCs](#throttling-retry-attempts-and-hedged-rpcs)
+        * [Pushback](#pushback)
+        * [Summary of Retry and Hedging Logic](#summary-of-retry-and-hedging-logic)
+     * [Retry Internals](#retry-internals)
+        * [Where Retries Occur](#where-retries-occur)
+        * [When Retries are Valid](#when-retries-are-valid)
+        * [Buffered RPCs](#buffered-rpcs)
+        * [Transparent Retries](#transparent-retries)
+        * [Exposed Retry Metadata](#exposed-retry-metadata)
+        * [Disabling Retries](#disabling-retries)
+        * [Retry and Hedging Statistics](#retry-and-hedging-statistics)
+     * [Configuration Language](#configuration-language)
+        * [Retry Policy](#retry-policy)
+        * [Hedging Policy](#hedging-policy-1)
+        * [Throttling Configuration](#throttling-configuration)
+        * [Integration with Service Config](#integration-with-service-config)
+  * [Open issues](#open-issues)
+     * [Security](#security)
+
+## Abstract
+
+gRPC client library will automatically retry failed RPCs according to a policy set by the service owner.
+
+## Background
+
+Currently, gRPC does not retry failed RPCs. All failed RPCs are immediately returned to the application layer by the gRPC client library.
+
+Many teams have implemented their own retry logic wrapped around gRPC like [Veneer Toolkit](https://github.com/googleapis/toolkit) and [Cloud Bigtable](https://github.com/GoogleCloudPlatform/cloud-bigtable-client).
+
+## Proposal
+
+### Overview
```

Comment:
Oh, one more thing: This section should probably include a link to https://github.com/grpc/proposal/pull/5.